e2e-tests/native/test/helpers/Gestures.ts

```diff
@@ -0,0 +1,179 @@
+ * screen based on the SCREEN_SIZE which will be determined once if needed
+ * multiple times.
+ */
+
+let SCREEN_SIZE: RectReturn | null
+type XY = {
+  x: number
+  y: number
+}
+
+/**
+ * The values in the below object are percentages of the screen
+ */
+const SWIPE_DIRECTION = {
+  down: {
+    start: { x: 50, y: 15 },
+    end: { x: 50, y: 85 }
+  },
+  left: {
+    start: { x: 95, y: 50 },
+    end: { x: 5, y: 50 }
+  },
+  right: {
+    start: { x: 5, y: 50 },
+    end: { x: 95, y: 50 }
+  },
+  up: {
+    start: { x: 50, y: 85 },
+    end: { x: 50, y: 15 }
+  }
+}
+
+const SWIPE_PERCENTAGE = 0.85
+const WAIT_FOR_SWIPE_FINISHED = 1000
+
+class Gestures {
+  /**
+   * Check if an element is visible and if not wipe up a portion of the screen to
+   * check if it visible after x amount of scrolls
+   */
+  static async checkIfDisplayedWithSwipeUp(
+    element: ChainablePromiseElement<Promise<WebdriverIO.Element>>,
+    maxScrolls: number,
+    amount = 0
+  ): Promise<void> {
+    // If the element is not displayed and we haven't scrolled the max amount of scrolls
+    // then scroll and execute the method again
+    if (!(await element.isDisplayed()) && amount <= maxScrolls) {
+      await this.swipeUp(SWIPE_PERCENTAGE)
+      await this.checkIfDisplayedWithSwipeUp(element, maxScrolls, amount + 1)
+    } else if (amount > maxScrolls) {
+      // If the element is still not visible after the max amount of scroll let it fail
+      throw new Error(`The element '${element}' could not be found or is not visible.`)
+    }
+
+    // The element was found, proceed with the next action
+  }
+
+  /**
+   * Swipe down based on a percentage
+   */
+  static async swipeDown(percentage = 1): Promise<void> {
+    await this.swipeOnPercentage(
+      this.calculateXY(SWIPE_DIRECTION.down.start, percentage),
+      this.calculateXY(SWIPE_DIRECTION.down.end, percentage)
+    )
+  }
+
+  /**
+   * Swipe Up based on a percentage
+   */
+  static async swipeUp(percentage = 1): Promise<void> {
+    await this.swipeOnPercentage(
+      this.calculateXY(SWIPE_DIRECTION.up.start, percentage),
+      this.calculateXY(SWIPE_DIRECTION.up.end, percentage)
+    )
+  }
+
+  /**
+   * Swipe left based on a percentage
+   */
+  static async swipeLeft(percentage = 1): Promise<void> {
+    await this.swipeOnPercentage(
+      this.calculateXY(SWIPE_DIRECTION.left.start, percentage),
+      this.calculateXY(SWIPE_DIRECTION.left.end, percentage)
+    )
+  }
+
+  /**
+   * Swipe right based on a percentage
+   */
+  static async swipeRight(percentage = 1): Promise<void> {
+    await this.swipeOnPercentage(
+      this.calculateXY(SWIPE_DIRECTION.right.start, percentage),
+      this.calculateXY(SWIPE_DIRECTION.right.end, percentage)
+    )
+  }
+
+  /**
+   * Swipe from coordinates (from) to the new coordinates (to). The given coordinates are
+   * percentages of the screen.
+   */
+  static async swipeOnPercentage(from: XY, to: XY): Promise<void> {
```

Comment:
Does this take percentages or coordinates? I think either the naming or the comment is wrong here. Perhaps it makes sense to just take a percentage and a `SWIPE_DIRECTION` as it is only used that way and calculate xy here

Reply:
from and two are both coordinates relative to the screen width and height. Which makes those coordinate percentages